src/js/components/MultiSelect/MultiSelect.js

```diff
@@ -0,0 +1,680 @@
+            Object.defineProperties(adjustedEvent, {
+              target: { value: inputRef.current },
+              value: { value: nextValue },
+              option: { value: option },
+            });
+          } else {
+            adjustedEvent = event;
+            adjustedEvent.target = inputRef.current;
+            adjustedEvent.value = nextValue;
+            adjustedEvent.option = option;
+          }
+          onChange(adjustedEvent);
+        }
+      },
+      [onChange, setValue, triggerChangeEvent],
+    );
+
+    const SelectIcon = getSelectIcon(icon, theme, open);
+
+    const optionValue = useCallback(
+      (index) => getOptionValue(index, allOptions, valueKey),
+      [allOptions, valueKey],
+    );
+
+    const isDisabled = useCallback(
+      (index) =>
+        checkDisabled(index, disabled, disabledKey, allOptions, valueKey),
+      [disabled, disabledKey, allOptions, valueKey],
+    );
+
+    const visibleValue = useCallback(
+      (i) => {
+        const iValue = valueKey && valueKey.reduce ? applyKey(i, valueKey) : i;
+        const indexOptions = allOptions.indexOf(i);
+        const iLabel = getOptionLabel(indexOptions, allOptions, labelKey);
+        if (value.indexOf(iValue) < 5) {
+          let child;
+          if (children) {
+            const optionDisabled = isDisabled(indexOptions);
+            const optionSelected = true;
+            const optionActive = false;
+            child = children(i, indexOptions, allOptions, {
+              active: optionActive,
+              disabled: optionDisabled,
+              selected: optionSelected,
+            });
+          }
+
+          return (
+            <SelectOption
+              role="option"
+              a11yTitle={
+                value.includes(iValue)
+                  ? `${iLabel} selected`
+                  : `${iLabel} not selected`
+              }
+              aria-setsize={value.length}
+              aria-posinset={value.indexOf(iValue)}
+              aria-selected={value.includes(iValue)}
+              aria-disabled={isDisabled(indexOptions)}
+              plain
+              hoverIndicator
+              fill="horizontal"
+              tabIndex="0"
+              onClick={(event) => {
+                if (!isDisabled(indexOptions)) {
+                  const intermediate = [...value];
+                  const index = value.indexOf(iValue);
+                  if (intermediate.includes(iValue)) {
+                    onSelectChange(event, {
+                      option: iValue,
+                      value: intermediate.filter((v) => v !== iValue),
+                    });
+                    if (index !== intermediate.length - 1) {
+                      let timer = 0;
+                      // if index is the last one visible
+                      // allow time for the next option to
+                      // become visible
+                      if (index === 4) timer = 200;
```

Comment:
Why 4? This feels brittle potentially

Reply:
I'm also not following why we need a timer -- I haven't seen this pattern before of using `getElementById` etc as we do below. Can you elaborate on the purpose of this?

Reply:
5 is the maximum number of items that should be shown at the input level based on the designs. I could pull this out as a prop though. For hpe I think we are always recommending this be set to 5, but I could see other users wanting to adjust. I'm unsure if I want to make this a regular prop or theme prop. Right now I'm leaning towards a regular prop but open to other opinions on this.

Reply:
I have the timer because if our focus is on the last item within the visible selections at the input level but there are still more options selected that are not being shown (because we only show 5 at a time), when we remove the last visible option, we have to wait until the next option is rendered and then once it is visible I move the focus to that option.

Reply:
I could see both a theme value as well as a property for this. I'd probably start with just the theme for now and only add the prop later down the road when a demand surface more.

Reply:
Added `selectMultiple.visibleInline` as a theme prop